dump/parser.go

```diff
@@ -19,28 +19,73 @@ var (
 type ParseHandler interface {
 	// Parse CHANGE MASTER TO MASTER_LOG_FILE=name, MASTER_LOG_POS=pos;
 	BinLog(name string, pos uint64) error
-
+	GtidSet(gtidsets string) error
 	Data(schema string, table string, values []string) error
 }
 
+var oneGtidExp *regexp.Regexp
+var mutilGtidStartExp *regexp.Regexp
+var midUuidSet *regexp.Regexp
+var endUuidSet *regexp.Regexp
 var binlogExp *regexp.Regexp
 var useExp *regexp.Regexp
 var valuesExp *regexp.Regexp
 
 func init() {
+	//SET @@GLOBAL.GTID_PURGED='1638041a-0457-11e9-bb9f-00505690b730:1-429405150';
+	oneGtidExp = regexp.MustCompile("SET @@GLOBAL.GTID_PURGED='(.+)'")
+
+	//SET @@GLOBAL.GTID_PURGED='071a84e8-b253-11e8-8472-005056a27e86:1-76,	// start of multi-gtid set
+	//2337be48-0456-11e9-bd1c-00505690543b:1-7,								// middle gtid set
+	//41d816cd-0455-11e9-be42-005056901a22:1-2';   							// end gtid set
+	mutilGtidStartExp = regexp.MustCompile("SET @@GLOBAL.GTID_PURGED='(.+),")
+	midUuidSet = regexp.MustCompile("(^\\w{8}(-\\w{4}){3}-\\w{12}:\\d+-\\d+),")
```

Comment:
what's meaning of `start`, `middle`, `end` gtid set?

Reply:
`SET @@GLOBAL.GTID_PURGED='071a84e8-b253-11e8-8472-005056a27e86:1-76,
2337be48-0456-11e9-bd1c-00505690543b:1-7,
41d816cd-0455-11e9-be42-005056901a22:1-2';`
GTID_PURGED might contains multiple gtid sets, so I use start, middle, end, these three regex to match these conditions.